File: modules/chat-core/src/main/java/com/shterneregen/securelan/chat/service/ChatClientService.java
package com.shterneregen.securelan.chat.service;

import com.shterneregen.securelan.common.model.rtc.RtcSignalEnvelope;

public interface ChatClientService {
    boolean connect(ChatClientConnectRequest request);
    void disconnect();
    void sendMessage(String text);
    void sendSignal(RtcSignalEnvelope signal);
    boolean isConnected();
}

File: modules/chat-core/src/main/kotlin/com/shterneregen/securelan/chat/service/ChatClientService.kt
package com.shterneregen.securelan.chat.service

import com.shterneregen.securelan.common.model.rtc.RtcSignalEnvelope

interface ChatClientService {
    fun connect(request: ChatClientConnectRequest): Boolean
    fun disconnect()
    fun sendMessage(text: String?)
    fun sendSignal(signal: RtcSignalEnvelope?)
    fun isConnected(): Boolean
}
